.github/workflows/test.yml

```diff
@@ -0,0 +1,55 @@
+name: run tests
+
+on:
+  pull_request:
+    branches:
+      - '*'
+
+  # make workflow "callable" by others
+  workflow_call:
+
+
+jobs:
+  test:
+    runs-on: ubuntu-latest
+
+    container:
```

Comment:
Instead of using the Dagster image as a "CI container" here, we could also just use `docker build`/`docker buildx build` and `docker run ... tests`.

This would have the advantage that we 1) install dependencies and 2) it runs the tests exactly in the same way like it will in production.

Reply:
You would use commands similar to those you use in your local shell:

1. Build the image using e.g. `docker buildx build -t some-tag ...`.
2. Run the tests within it using e.g. `docker run --rm -i some-tag ...`.